File: src/main/java/ludwig/workspace/Environment.java
package ludwig.workspace;

public class Environment {
    private final Workspace workspace = new Workspace();
    private final UsageTracker usageTracker;
    private final SymbolRegistry symbolRegistry;

    public Environment() {
        symbolRegistry = new SymbolRegistry(workspace);
        usageTracker = new UsageTracker(workspace);
        workspace.init();
    }

    public Workspace workspace() {
        return workspace;
    }

    public UsageTracker usageTracker() {
        return usageTracker;
    }

    public SymbolRegistry symbolRegistry() {
        return symbolRegistry;
    }
}

File: src/main/java/ludwig/workspace/Environment.kt
package ludwig.workspace

class Environment {
    private val workspace = Workspace()
    private val usageTracker: UsageTracker
    private val symbolRegistry: SymbolRegistry

    init {
        symbolRegistry = SymbolRegistry(workspace)
        usageTracker = UsageTracker(workspace)
        workspace.init()
    }

    fun workspace(): Workspace {
        return workspace
    }

    fun usageTracker(): UsageTracker {
        return usageTracker
    }

    fun symbolRegistry(): SymbolRegistry {
        return symbolRegistry
    }
}
